File: src/test/java/libldt3/model/regel/F001Test.java
package libldt3.model.regel;

import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.IsEqual.equalTo;

public class F001Test {

	private static final F001 f001 = new F001();

	@Test
	public void testEmpty() {
		assertThat(f001.isValid(""), equalTo(false));
	}

	@Test
	public void testNull() {
		assertThat(f001.isValid(null), equalTo(true));
	}

	@Test
	public void testFiveZeros() {
		assertThat(f001.isValid("00000"), equalTo(true));
	}

	@Test
	public void testFiveNines() {
		assertThat(f001.isValid("99999"), equalTo(true));
	}

	@Test
	public void testOneTwoThreeFourFive() {
		assertThat(f001.isValid("12345"), equalTo(true));
	}

	@Test
	public void testOneTwoThreeFour() {
		assertThat(f001.isValid("1234"), equalTo(false));
	}

	@Test
	public void testOneTwoThreeFourFiveSix() {
		assertThat(f001.isValid("123456"), equalTo(false));
	}

	@Test
	public void testABCDE() {
		assertThat(f001.isValid("ABCDE"), equalTo(false));
	}

	@Test
	public void testFiveSymbols() {
		assertThat(f001.isValid("/=()["), equalTo(false));
	}

	@Test
	public void testOneTwoThreeFourA() {
		assertThat(f001.isValid("1234A"), equalTo(false));
	}

	@Test
	public void testOneTwoThreeFourLowerA() {
		assertThat(f001.isValid("1234a"), equalTo(false));
	}

}

File: cs/libldt3.tests/libldt3/model/regel/F001Test.cs
using NUnit.Framework;

namespace libldt3
{
    namespace model
    {
        namespace regel
        {
            [TestFixture]
            public class F001Test
            {
                static readonly F001 f001 = new F001();

                [Test]
                public void testEmpty()
                {
                    Assert.That(f001.IsValid(""), Is.False);
                }

                [Test]
                public void testNull()
                {
                    Assert.That(f001.IsValid(null), Is.True);
                }

                [Test]
                public void testFiveZeros()
                {
                    Assert.That(f001.IsValid("00000"), Is.True);
                }

                [Test]
                public void testFiveNines()
                {
                    Assert.That(f001.IsValid("99999"), Is.True);
                }

                [Test]
                public void testOneTwoThreeFourFive()
                {
                    Assert.That(f001.IsValid("12345"), Is.True);
                }

                [Test]
                public void testOneTwoThreeFour()
                {
                    Assert.That(f001.IsValid("1234"), Is.False);
                }

                [Test]
                public void testOneTwoThreeFourFiveSix()
                {
                    Assert.That(f001.IsValid("123456"), Is.False);
                }

                [Test]
                public void testABCDE()
                {
                    Assert.That(f001.IsValid("ABCDE"), Is.False);
                }

                [Test]
                public void testFiveSymbols()
                {
                    Assert.That(f001.IsValid("/=()["), Is.False);
                }

                [Test]
                public void testOneTwoThreeFourA()
                {
                    Assert.That(f001.IsValid("1234A"), Is.False);
                }

                [Test]
                public void testOneTwoThreeFourLowerA()
                {
                    Assert.That(f001.IsValid("1234a"), Is.False);
                }
            }
        }
    }
}
